src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs

```diff
@@ -96,26 +96,20 @@ public static FormattableString GetConversionFormattable(this Parameter paramete
         {
             if (toType.EqualsIgnoreNullable(Configuration.ApiTypes.RequestContentType))
             {
-                switch (parameter.Type)
+                if (parameter.Type.IsFrameworkType)
```

Comment:
why revert the pattern matching? It seems the if-else can be done using pattern matching.

Reply:
We need to check whether the type is `string`. switch not compiles on `typeof(string)`

Reply:
but when supports that.
You just need:
```csharp
case { IsFrameworkType: true } when parameter.Type.FrameworkType == typeof(string):
```

Reply:
or 
```csharp
case {} when parameter.Type.EqualsIgnoreNullable(typeof(string)):
```